### PR TITLE
#10193: Issue with WFS added via Query params

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR, the issue of not adding WFS layer to map and be not visible is resolved. GetFeature request is performed now correctly by MS.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#10193 

**What is the current behavior?**
#10193 

**What is the new behavior?**
WFS layer is added to map using query params in url if the url contains queryString.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information

### CODE REVIEW

File: web/client/api/WFS.js [hunk 1]
```diff
@@ -12,6 +12,13 @@ import requestBuilder from '../utils/ogc/WFS/RequestBuilder';
 import {toOGCFilterParts} from '../utils/FilterUtils';
 import { getDefaultUrl } from '../utils/URLUtils';
 
+/**
+* @note from this ref: https://www.npmjs.com/package/url#urlformaturlobj
+* * in using [ url.format(urlObj) ] --> 'urlObj.search' will be used in place of 'urlObj.query'
+* * and 'urlObj.query' (object; see querystring) will only be used if 'urlObj.search' is absent.
+* * so in case adding new parameters or update existing ones in 'urlObj.query', we need to set 'urlObj.search' with undefined to ignore it
+*/
+
```
Comment (dsuren1) [inline web/client/api/WFS.js:21]:
```suggestion
```

I don't think this comment is needed as it is not a new feature. Inline comment on the respective functions is fine

**Review by dsuren1 — CHANGES_REQUESTED** [review]

**Review by dsuren1 — APPROVED** [review]

**Comment by dsuren1** [comment]
@ElenaGallo Kindly test it in DEV and let us know if it can be backported. Thanks!

**Comment by tdipisa** [comment]
@ElenaGallo @mahmoudadel54 this is for 2024.01.01 therefore let's wait the delivery of 2024.01.00 before backporting this. Thank you.

**Comment by ElenaGallo** [comment]
Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks

**Comment by mahmoudadel54** [comment]
> Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks
@ElenaGallo 
Backport is Done here ---> https://github.com/geosolutions-it/MapStore2/pull/10296

**Pull request merged**